ast/src/parsed/asm.rs

```diff
@@ -167,6 +167,17 @@ impl SymbolPath {
         })
     }
 
+    /// Returns the path up to the last part and the last part individually.
+    /// Returns None if the last part is `super` or the path is empty.
+    pub fn try_split_off_last_part(&self) -> Option<(Self, String)> {
```

Comment:
i think this could be just called `pop` or `pop_last` and take `&mut self`.
Similar to `PathBuf` and `Vec`.